### DISCUSSION TITLE
Unresponsive behavior using WebGpu simple example and device warning

### DISCUSSION THREAD

**Discussion by sameer-dudeja** [gpuweb/gpuweb#4106] [Q&A] [answered]
![image](https://github.com/gpuweb/gpuweb/assets/16884060/2f846c56-49dd-48bb-aaad-b9b82e1ab217)
````
       const canvas = document.querySelector("canvas");
        if (!navigator.gpu) {
            throw new Error("WebGPU not supported on this browser.");
        }
        const adapter = await navigator.gpu.requestAdapter();
        if (!adapter) {
            throw new Error("No appropriate GPUAdapter found");
        }
        const device = await adapter.requestDevice();
        const context = canvas.getContext("webgpu");
        const canvasFormat = navigator.gpu.getPreferredCanvasFormat();
        context.configure({
            device: device,
            format: canvasFormat
        })

        const encoder = device.createCommandEncoder();

        for (var i = 0; i < 100000; i++) {
            var pass = encoder.beginRenderPass({
                colorAttachments: [{
                    view: context.getCurrentTexture().createView(),
                    loadOp: "clear",
                    clearValue: { r: Math.random(), g: Math.random(), b: Math.random(), a: 1.0 },
                    storeOp: "store",
                }]
            });
            console.log(pass)
            pass.end();
        }

        device.queue.submit([encoder.finish()]);
`

**Comment by kainino0x** [comment, marked as answer]
(This probably isn't an issue with the WebGPU specification so I've converted it to a discussion thread. Browser bugs - which I don't think this is - should be filed against the appropriate browser.)

I think the problem here is the command encoder is simply too large. It's not typical to use many render passes in a single frame; if you're drawing a lot of objects it's much more efficient to do so inside few render passes (1 in the basic case). Device loss like this is normal and expected in cases where you push too far, either with too large/too many resources, too many draws/commands, too slow shaders, etc.

**Comment by greggman** [comment]
This appears to be an issue with Chrome, not the spec. This site is for the spec, not specific browsers. You can file a bug at http://crbug.com for chrome bugs.
